stringer.go

```diff
@@ -0,0 +1,30 @@
+package dig
+
+import (
+	"bytes"
+	"fmt"
+)
+
+// String representation of the entire Container
+func (c *Container) String() string {
```

Comment:
This functionality is very self-contained even though it belongs to the same package.
I'd like to keep `dig.go` as tight as possible -- just the core functionality.